go/stream/message_test.go

```diff
@@ -281,6 +281,65 @@ func TestMessage(t *testing.T) {
 		require.EqualError(t, err, "Key: 'Message.Properties.WorkspaceID' Error:Field validation for 'WorkspaceID' failed on the 'required' tag")
 	})
 
+	t.Run("validation Err: with encryption properties", func(t *testing.T) {
+		validator := stream.NewMessagePropertiesValidator(stream.WithEncryptionPropertiesValidator())
+		msg := stream.Message{
+			Properties: stream.MessageProperties{
+				RequestType: "requestType",
+				RoutingKey:  "routingKey",
+				WorkspaceID: "workspace-id",
+				SourceID:    "sourceID",
+				RequestIP:   "10.29.13.20",
+				ReceivedAt:  time.Date(2024, 8, 1, 0o2, 30, 50, 200, time.UTC),
+				Encryption:  "some-serialized-encryption-settings",
+			},
+			Payload: json.RawMessage(`{}`),
+		}
+
+		err := validator(&msg.Properties)
+		require.EqualError(t, err, "encryption key ID is required when encryption is set")
+	})
+
+	t.Run("validation ok: without encryption properties", func(t *testing.T) {
+		validator := stream.NewMessagePropertiesValidator()
+		msg := stream.Message{
+			Properties: stream.MessageProperties{
+				RequestType: "requestType",
+				RoutingKey:  "routingKey",
+				WorkspaceID: "workspace-id",
+				SourceID:    "sourceID",
+				RequestIP:   "10.29.13.20",
+				ReceivedAt:  time.Date(2024, 8, 1, 0o2, 30, 50, 200, time.UTC),
+				Encryption:  "some-serialized-encryption-settings",
+			},
+			Payload: json.RawMessage(`{}`),
+		}
+
+		err := validator(&msg.Properties)
+		require.NoError(t, err)
+	})
+
+	t.Run("validation ok", func(t *testing.T) {
```

Comment:
We already have a "validation ok" test at the same level, right? Should we call this one `"validation ok: encryption"`?